ocaml/idl/datamodel_cluster.ml

```diff
@@ -152,6 +152,10 @@ let t =
           ~ty:(Set (Ref _cluster_host)) "cluster_hosts"
           "A list of the cluster_host objects associated with the Cluster"
 
+      ; field ~qualifier:DynamicRO ~lifecycle:[ Prototyped, rel_lima, "" ]
+          ~ty:(Set String) "pending_forget" ~default_value:(Some (VSet []))
+          "A list of cluster members that were declared permanently dead and not yet removed"
+
```

Comment:
Should we set `~internal_only:true` to make this field invisible to the user, since the API call will be hidden too?

Reply:
Actually that doesn't work, it would hide the field from records.ml too and the API, and we need it visible from the CLI for debugging.

Reply:
Hmm, I've asked the others. It would still be nice to make it `internal_only` somehow, so this does not show up to the user as a new field. But I guess it's not a disaster if it does, because it still conveys some useful information to them.

Reply:
You can at least mention in the documentation that this is not meant to be used and it is for internal use only (like VM.create for example)

Reply:
Added a fixup to mention this is an internal field, it is RO so the user won't be able to change it. And in fact its not bad that they can see it: the admin asked one host to be destroyed (permanently dead), but it is still part of the cluster as far as quorum is concerned until he declares all the unreachable hosts as permanently dead.
(to declare a host as permanently dead all the other members of the cluster must confirm, which they can't do if they're offline).

So if the admin wants to find out who are the members of the cluster at a particular moment in time they can look at the "cluster_hosts" field plus the "pending_forget" field. (the host/pif objects will be destroyed so if we didn't preserve the IP in pending_forget there would be no way to find out). (or they can run the low-level diagnostic commands from the clustering daemon)

Reply:
Why do you need to add a manual newline there? I think we don't use it anywhere else in the messages

Reply:
Indeed I just need a \ to join the strings but not an actual newline in the output, dropped.